### PR TITLE
change print to logging

### LINKED ISSUE #282: Switch from print to logging
Use logging instead of print.   (@Ir1d you brought this up before)

Suggested by @adefazio

### PR DESCRIPTION
**Pull request by Ir1d**
# Before submitting

- Was this discussed/approved via a Github issue? (no need for typos, doc improvements) Should close #282 
- Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?  Y
- Did you make sure to update the docs?   Y
- Did you write any new necessary tests?  N

## What does this PR do?
Fixes #282  (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.   

## Did you have fun?
Make sure you had fun coding 🙃

## Note

I changed `print` to `logging.info`. And added some simple support for Custom Logger to override the logging to some custom logging functions.
However, many logging functions dont have access to trainer.logger at this moment, so I leaved them untouched.(for example the callback has test_tube_logger, some mixin functions have unclear direct access to logger. In these cases, I let them use logging.info directly instead of using a more robust self.logger.info)

### CODE REVIEW

File: pl_examples/basic_examples/lightning_module_template.py [hunk 1]
```diff
@@ -214,17 +214,17 @@ def __dataloader(self, train):
 
     @pl.data_loader
     def train_dataloader(self):
-        print('training data loader called')
+        self.logger.info('training data loader called')
```
Comment (Borda) [inline pl_examples/basic_examples/lightning_module_template.py]: this is not very saved with `self.logger = None` in `LightningModule`

Comment (Ir1d) [reply pl_examples/basic_examples/lightning_module_template.py]: ah indeed. in that case exposing the logger is becoming much harder 🌚

Comment (Borda) [reply pl_examples/basic_examples/lightning_module_template.py]: You may try change `self.logger = logging.getLogger()`

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 2]
```diff
@@ -121,10 +122,10 @@ def on_epoch_end(self, epoch, logs=None):
         current = logs.get(self.monitor)
         stop_training = False
         if current is None:
-            logging.info(
-                'Early stopping conditioned on metric `%s` '
-                'which is not available. Available metrics are: %s' %
-                (self.monitor, ','.join(list(logs.keys()))), RuntimeWarning)
+            warnings.warn(
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: `warnings.warn` and `logging.warning` are a completely different stream with different consequences so please be sure that you want this behaviour... You may consider also raise it as Error

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 3]
```diff
@@ -121,10 +122,10 @@ def on_epoch_end(self, epoch, logs=None):
...
+                f'Early stopping conditioned on metric `{self.monitor}` '
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: it is recommended not having white space on the end of the line but rather on the beginning of the following one...

File: pytorch_lightning/callbacks/pt_callbacks.py [hunk 4]
```diff
@@ -142,7 +143,7 @@ def on_epoch_end(self, epoch, logs=None):
 
     def on_train_end(self, logs=None):
         if self.stopped_epoch > 0 and self.verbose > 0:
-            logging.info('Epoch %05d: early stopping' % (self.stopped_epoch + 1))
+            logging.info(f'Epoch {self.stopped_epoch + 1:05d}: early stopping')
```
Comment (Borda) [inline pytorch_lightning/callbacks/pt_callbacks.py]: not directly to this line, but we shall be sure that all this new logging is tested (covered by tests)

File: pytorch_lightning/callbacks/pt_callbacks.py (hunk 2 above)
Comment (Ir1d) [reply pytorch_lightning/callbacks/pt_callbacks.py]: hmmm I'm not familiar with this, but I noticed that the previous code used warnings.warn a lot, I thought this was kind of best-practice.

File: pytorch_lightning/callbacks/pt_callbacks.py (hunk 4 above)
Comment (Ir1d) [reply pytorch_lightning/callbacks/pt_callbacks.py]: This did occur to me, but I didn't know how to test this logging, any idea on this?

File: pytorch_lightning/callbacks/pt_callbacks.py (hunk 3 above)
Comment (Ir1d) [reply pytorch_lightning/callbacks/pt_callbacks.py]: lol it passed flake8. but I'll change this.

File: pytorch_lightning/callbacks/pt_callbacks.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/callbacks/pt_callbacks.py]: I believe that it was related to using `print`, in such case, it is better to `warnings`
I do not say that is wrong, in this case it is good, just to be aware of it... :]

**Comment by Ir1d** [comment]
A summary of this PR:
1. changed print -> logging.info and warnings.warn
2. set logging options [here](https://github.com/williamFalcon/pytorch-lightning/pull/457/files#diff-0cd7170e96efa4a55577086f5db2fe4cR239), but not sure if appropiate. @williamFalcon you might want to take a close look.
3. no new tests included, because I didn't know how to test logging :|

oh and btw, I thought it might be better to leave 'custom logging function' for some other day in the future. At this moment, I don't feel like adding a property to trainer indicating logging.info . The logging conventions in the project seems not yet settled down.

File: pytorch_lightning/callbacks/pt_callbacks.py (hunk 4 above)
Comment (Borda) [reply pytorch_lightning/callbacks/pt_callbacks.py]: I usually tested that the message does not crash the code...

**Comment by williamFalcon** [comment]
amazing!

@Borda let's do a different PR/issue for test coverage.

**Pull request merged**